### ISSUE TITLE
How do I generate a for loop?

### ISSUE THREAD

**Issue by nerndt** [FlorianRappl/Mages#86]
Thanks for the wonderful project Florian!
 I modified the Sample solution and use a multiline richtextbox instead of a textbox.  I added an 'interpret' button to process each line one at a time so I can highlight the error and explain the issue.

I tried a variety of for examples with no luck:

for (i = 0; i < 10; i++)

What is the correct syntax?

Thanks, Nick

**Comment by FlorianRappl** [comment]
Hi Nick, thanks for the interest in the project and the nice words.

The syntax you are using seems to be correct. I just ran MAGES from the console and got the following output.

```c
SWM> for (i = 0; i < 5; i++) { console.write(i);  }
012344
```

(The additional 4 is due to the MAGES console printing the top value of the stack.)

What kind of errors are you receiving?

**Comment by nerndt** [comment]
Thanks Florian.
When I place the code above across multiple lines I get the error: "Error interpreting console line"
I will have for loops that will cover multiple lines (be pretty complex). I would like to be able to break more complex lines into individual lines so I can specifically tell the user where they make a syntax mistake.
Also, do you have example scripts that show all of your syntax (how to do a for, while, etc.)?
Take care, Nick

**Comment by nerndt** [comment]
To get the Error I mentioned, I broke your line into the following:
for (i = 0; i < 5; i++)
{
console.write(i);
}

**Comment by FlorianRappl** [comment]
Ah, yeah, I think this could indeed be a problem / bug that should be addressed. But, just to be sure - the following works, right?

```c
for (i = 0; i < 5; i++) {
  console.write(i);
}
```

(i.e., the issue is with newlines / follow up after the for-loop head.)

**Comment by FlorianRappl** [comment]
Regarding the examples: Most of the available examples are in the docs (https://github.com/FlorianRappl/Mages/tree/master/doc), but I agree that it is definitely not enough. I will try to find some time soon to fix the potential bug and improve the documentation. Thanks for pointing out the flaws!

**Comment by nerndt** [comment]
Hi Florian.
I am afraid not!


 Take care, Nick Nick Erndt nerndt@yahoo.com hm 650 493-7607 mobile 650 906-5970

**Comment by FlorianRappl** [comment]
I will close this issue - the "bug" addressed here is just wrong usage (of the REPL). The REPL cannot know when a statement ends, but it uses some trivial techniques, which will not end the statement in some cases. However, these techniques do not apply to `for (i = 0, i < 5; i++)`. I will introduce a shift + enter combination that will always lead to a new line instead of triggering the evaluation.

TL;DR: Only a problem with the REPL and only if no indicator of an unfinished statement is given.
